library/src/main/java/com/vinted/coper/CoperImpl.kt

```diff
@@ -59,6 +72,52 @@ internal class CoperImpl(private val fragmentManager: FragmentManager) : Coper {
         }
     }
 
+    internal suspend fun getFragmentSafely(): CoperFragment {
+        val deferred = CompletableDeferred<CoperFragment>()
+        val fragment = getAttachedFragment()
+        fragment.lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                deferred.complete(fragment)
+            }
+        })
```

Comment:
Same about safety. Maybe you should start waiting for this callback before adding `Fragment` to fragment manager?

Reply:
same as above

Reply:
I have some concern as Gediminas and `same as above` does not explain your motivation here

Reply:
Oh, I will check if I can reach lifecycle before fragment is attached

Reply:
also, why to wait after that, lifecycle observer will throw all lifecycle events by documentation, added test for that with activity